apps/condo/domains/common/constants/import.js

```diff
@@ -35,17 +39,34 @@ const DATE_TIME_FORMATS = DATE_FORMATS
         TIME_VARIANTS.map(timeVariant => [dateFormat, timeVariant].join(' '))
     )
 
+/** @param {dayjs.Dayjs} date */
+function yearAroundToday (date) {
+    return Math.abs(date.year() - dayjs().year()) < VALID_YEAR_OFFSET_FROM_TODAY
+}
+
+/** @type {(string | {format:string, validate:(validDate: dayjs.Dayjs)=>boolean})[]} */
 const DEFAULT_DATE_PARSING_FORMATS = [
     ...DATE_TIME_FORMATS,
     ...DATE_FORMATS,
 
+    { format: 'YYYYMM', validate: yearAroundToday },
+    { format: 'MMYYYY', validate: yearAroundToday },
+
     'YYYY-MM-DDTHH:mm:ss.SSS[Z]', // The result of dayjs().toISOString()
     'YYYY-MM-DDTHH:mm:ss.SSSZ',
     'YYYY-MM-DDTHH:mm:ss.SSS',
     'YYYY-MM-DDTHH:mm:ssZZ',
     'YYYY-MM-DDTHH:mm:ssZ',
     'YYYY-MM-DDTHH:mm:ss',
-].sort((a, b) => b.length - a.length) // Order matters! see "Differences to moment" https://day.js.org/docs/en/parse/string-format
+].sort((a, b) => {
+    if (typeof a === 'string' && typeof b === 'string') {
+        return b.length - a.length
+    }
+    if (typeof a === 'string' || typeof b === 'string') {
```

Comment:
What's this?

Reply:
Why not to compare a or a.format?

Reply:
Need to sort formats from longest to shortest.
Formats are: strings, objects with "format" property